File: AppGenerateFiles/dbo/Model/Asignaturas.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Asignaturas : EntityClass {
       [PrimaryKey(Identity = true)]
       public int? Id { get; set; }
       public string? Nombre { get; set; }
       public string? Nombre_corto { get; set; }
       public string? Observaciones { get; set; }
       public int? Nivel_id { get; set; }
       public bool? Habilitado { get; set; }
       public DateTime? Created_at { get; set; }
       public DateTime? Updated_at { get; set; }
       public int? Orden { get; set; }
       [ManyToOne(TableName = "Niveles", KeyColumn = "Id", ForeignKeyColumn = "Nivel_id")]
       public Niveles? Niveles { get; set; }
       [OneToMany(TableName = "Docente_asignaturas", KeyColumn = "Id", ForeignKeyColumn = "Asignatura_id")]
       public List<Docente_asignaturas>? Docente_asignaturas { get; set; }
       [OneToMany(TableName = "Materias", KeyColumn = "Id", ForeignKeyColumn = "Asignatura_id")]
       public List<Materias>? Materias { get; set; }
   }
}

File: UI/wwwroot/Model/Asignaturas.js
//@ts-check
import { Niveles }  from './Niveles.js'
import { Docente_asignaturas }  from './Docente_asignaturas.js'
import { Materias }  from './Materias.js'
import { EntityClass } from '../WDevCore/WModules/EntityClass.js';
class Asignaturas extends EntityClass    {
   /** @param {Partial<Asignaturas>} [props] */
   constructor(props) {
       super(props, 'EntityDbo');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id;
   /**@type {String}*/ Nombre;
   /**@type {String}*/ Nombre_corto;
   /**@type {String}*/ Observaciones;
   /**@type {Boolean}*/ Habilitado;
   /**@type {Date}*/ Created_at;
   /**@type {Date}*/ Updated_at;
   /**@type {Number}*/ Orden;
   /**@type {Niveles} ManyToOne*/ Niveles;
   /**@type {Array<Docente_asignaturas>} OneToMany*/ Docente_asignaturas;
   /**@type {Array<Materias>} OneToMany*/ Materias;
}
export { Asignaturas }
